File: frontend/src/app/shared/models/order.ts
import { Observable } from 'rxjs';
import { Product } from './product.model';

export enum OrderStatus {
  PENDING = 'pending',
  CONFIRMED = 'confirmed',
  COMPLETED = 'completed',
  CANCELLED = 'canceled',
}

export interface Order {
  id: string;
  products: {
    id: string;
    quantity: number;
    info: Product;
  }[];
  buyer: string;
  seller: string;
  totalPrice: number;
  status: OrderStatus;
  createdAt: Date;
  orderHistoryName: string;
  buyerName: Observable<string>;
  sellerName: Observable<string>;
}

File: backend/services/orders/src/main/java/pw/mer/letsplay/model/Order.java
package pw.mer.letsplay.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Getter
@Document("Order")
@JsonIgnoreProperties(ignoreUnknown = true)
public class Order {
    @Id
    @JsonProperty("id")
    private String id;

    @JsonProperty("buyer")
    private String buyer;

    @JsonProperty("seller")
    private String seller;

    @JsonProperty("products")
    private List<Product> products;

    @JsonProperty("totalPrice")
    private Double totalPrice;

    @JsonProperty("status")
    @Setter
    private EStatus status;

    @JsonProperty("createdAt")
    private Long createdAt;

    public Order(String buyer, String seller, List<Product> products, Double totalPrice) {
        this.buyer = buyer;
        this.seller = seller;
        this.products = products;
        this.totalPrice = totalPrice;
        this.status = EStatus.PENDING;

        this.createdAt = System.currentTimeMillis();
    }
}
